### PR TITLE
Add order descriptions via order item index 

### LINKED ISSUE #87: Add order descriptions that are already in the order item model
Update relevant command parsers
* If oo/INDEX is used, add order item at specified index as an order description
* If o/ is used to used, add that order description to the order and add that to the order item model (unless it already exists in the order item model)
* if order descriptions are edited, the editing behaviour might be the same as tags e.g. editing replaces all existing order descriptions with a new order description?

### PR DESCRIPTION
**Pull request by pPris**
Mostly ready for review, just cleaning up comments 

Can mix and match o/ and oi/ as long as at least one is present 

E.g.
- add  p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 **`o/Chocolate Cakes o/2 oi/2`** n/Dina t/friends t/owesMoney d/01/01/2022
 
- add  p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 **`oi/2`** n/Dyna t/friends t/owesMoney d/01/01/2022
 
- add  p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 **`o/chocolate cake`** n/Deeena /friends t/friendly d/01/01/2022

Had quite a lot of fake data and temporary stuff before the storage pr was done - it should be removed soon

### CODE REVIEW

File: src/main/java/seedu/cakecollate/commons/core/Messages.java [hunk 1]
```diff
@@ -11,6 +11,8 @@
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
     public static final String MESSAGE_INVALID_ORDER_DISPLAYED_INDEX = "The order index provided is invalid";
+    public static final String MESSAGE_INVALID_ORDER_ITEM_INDEX = "The order item index provided is invalid";
+    // shouldn't all these be more descriptive...
```
Comment (simran-bhadani3) [inline src/main/java/seedu/cakecollate/commons/core/Messages.java:15]: Actually true, I guess we can work on making them more descriptive in v1.4

File: src/main/java/seedu/cakecollate/commons/core/index/IndexList.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ public void add(Index index) {
     }
 
     public List<Index> getIndexList() {
-        return this.indexList;
+        return this.indexList; // unmodifiable?
```
Comment (simran-bhadani3) [inline src/main/java/seedu/cakecollate/commons/core/index/IndexList.java:26]: Should be modifiable.

File: src/main/java/seedu/cakecollate/commons/core/index/IndexList.java [hunk 3]
```diff
@@ -32,6 +32,15 @@ public void add(Index index) {
...
     public boolean checkIfEqual(IndexList other) {
         boolean equal = true;
+
+        if (other == null) {
+            return this == null;
+        }
+
+        // will this affect anyone else's work
+        this.sortList();
+        other.sortList();
```
Comment (simran-bhadani3) [inline src/main/java/seedu/cakecollate/commons/core/index/IndexList.java:42]: Good catch! I didn't think about sorting them first.

**Review by simran-bhadani3 — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/cakecollate/logic/commands/AddCommand.java [hunk 4]
```diff
@@ -65,10 +108,241 @@ public CommandResult execute(Model model) throws CommandException {
...
+     * Adds order descriptions in the descriptor to order items model if they aren't already in the model.
+     * These order descriptions are inputted by the user and already added into the descriptor before
+     * this add command object is created.
+     *
+     * @param model
+     */
+    private void addToOrderItems(Model model) {
+        this.addOrderDescriptor.getOrderDescriptions().get().stream()
+                .map(OrderDescription::getValue) // because a string is needed for creating a new Type for new OrderItem
+                .map(o -> new OrderItem(new Type(o))) // map to order item so can check if already in model
```
Comment (pavz02) [inline src/main/java/seedu/cakecollate/logic/commands/AddCommand.java:121]: Maybe we can consider taking out cost from order item entirely.

File: src/main/java/seedu/cakecollate/model/util/SampleDataUtil.java [hunk 5]
```diff
@@ -37,12 +37,12 @@
         return new Order[] {
             new Order(new Name("Alex Yeoh"), new Phone("87438807"), new Email("alexyeoh@example.com"),
                 new Address("Blk 30 Geylang Street 29, #06-40"),
-                    getOrderDescriptionSet("2 x Chocolate Cake"), getTagSet("friends"),
```
Comment (pavz02) [inline src/main/java/seedu/cakecollate/model/util/SampleDataUtil.java]: Any reason why you took out the number part haha

**Review by pavz02 — COMMENTED** [review]
Looks good!!

File: src/main/java/seedu/cakecollate/model/util/SampleDataUtil.java (hunk 5 above)
Comment (pPris) [reply src/main/java/seedu/cakecollate/model/util/SampleDataUtil.java]: Ahh I wasn't sure if we wanted the users to type in numbers, or if we should be giving them another way to add multiple of each item at the same time

Comment (pPris) [reply src/main/java/seedu/cakecollate/model/util/SampleDataUtil.java]: Reverted. My previous reply is irrelevant till v1.4 haha

**Pull request merged**